doc/scenario.md

```diff
@@ -0,0 +1,394 @@
+from scenario:
+```yaml
+# edit_topology_playbook.yml
+
+- name: Edit topology of my cluster
+  hosts: all
+  tasks:
+    - name: Edit topology by core 1
+      import_role:
+        name: tarantool.cartridge
+      vars:
+        cartridge_scenario:
+          - edit_topology
+        control_instance:
+          name: core_1
+          console_sock: '/var/run/tarantool/core_1.control'
+```
+
+### Replace delivering package to remote host
+
+By default, package specified by `cartridge_package_path` is simple copied to remote host, it's happened
+on `deliver_package` step. Sometimes we need to download or copy package from some custom place. Let's do it!
+
+Generally, we just need to override default `deliver_package` step.
+
+Let's take a look on it's [API](#deliver_package). It requires two variables:
+- `cartridge_package_path` that we simply specify in vars;
+- `single_instances_for_each_machine` that allows us to run this task once per each machine.
+
+As a result of this module we should set `delivered_package_path` variable
+(a path of package on remote machine).
+
+Now, choose a directory where our custom steps are placed, for example `./custom_steps`.
+
+Then create `./custom_steps/deliver_package.yml` and describe a way to deliver package:
+```yaml
+# ./custom_steps/deliver_package.yml
+
+- when:
+    - cartridge_package_path is not none
+    - inventory_hostname in single_instances_for_each_machine
+  tags: cartridge-instances
+  block:
+    - name: 'Download package'
+      get_url:
+        url: 'https://some-server/myapp-1.0.0-0.rpm'
+        dest: '/tmp'
+      register: downloaded_package
+
+    - name: 'Set delivered package path'
+      set_fact:
+        delivered_package_path: '{{ downloaded_package.dest }}'
+```
+
+Import role and say where to find our custom steps:
+```yaml
+# deploy_application.yml
+
+- name: My playbook
+  hosts: all
+  tasks:
+    - name: Deploy application
+      import_role:
+        name: tarantool.cartridge
+      vars:
+        cartridge_custom_steps_dir: "./custom_steps"
+```
+
+### Adding custom step to scenario
+
+You can create your own step and include it to scenario. Let's create `./custom_steps/special_magic.yml`. After that,
+you can add `special_magic` step to scenario. Just pass path to a directory where your custom steps are placed:
+
+```yaml
+- name: My playbook
+  hosts: all
+  tasks:
+    - name: Deploy application
+      import_role:
+        name: tarantool.cartridge
+      vars:
+        cartridge_custom_steps_dir: "./custom_steps"
+        cartridge_scenario:
+          - restart_instance
+          - special_magic
+```
+
+### Importing steps from different directories
+
+If you want to import steps by full paths, `cartridge_custom_steps` parameter can be useful. It allows specifying
+mapping between step name and path to file to import:
+
+```yaml
+cartridge_custom_steps:
+  - name: 'common_special_task'
+    file: './common/special_task.yml'
+```
+
```

Comment:
I think we can place `Adding custom step to scenario` and `Importing steps from different directories` at the beginning of examples and little rescturture them.

###  Adding custom step to scenario
...
### Importing steps from different directories
...
### Override task from default scenario

Sometimes we need to use default scenario, but change some steps realization. 
For example, sometimes we need to deliver package to remote our own way. .....

### Edit topology without connecting to membership

Now there is a big problem on deploying bug clusters - `connect_to_membership` step is too long.
Using scenario, we can solve this problem until it isn't solved in `cartridge`.

In fact `connect_to_membership` is used to find some instance that is already in cluster.
This instance should be used for joining other instances (otherwise two different clusters are created).
This instance is called `constol_instance` (link) and is used for editing topology and configuring cluster (auth, config and so on).
Generally, `connect_to_membership` step can be skipped if you definitely know some instance that is already joined to cluster. The solution is to set `control_instance` fact manually and remove `connect_to_membership` step from scenario.

.....